csharp/src/Ice/IceDiscovery/Plugin.cs

```diff
@@ -42,16 +42,15 @@ public async ValueTask DisposeAsync()
 
         public void Initialize(PluginInitializationContext context)
         {
-            bool ipv4 = _communicator.GetPropertyAsBool("Ice.IPv4") ?? true;
             bool preferIPv6 = _communicator.GetPropertyAsBool("Ice.PreferIPv6Address") ?? false;
             string address;
-            if (ipv4 && !preferIPv6)
+            if (preferIPv6)
             {
-                address = _communicator.GetProperty("IceDiscovery.Address") ?? "239.255.0.1";
+                address = _communicator.GetProperty("IceDiscovery.Address") ?? "ff15::1";
             }
             else
             {
-                address = _communicator.GetProperty("IceDiscovery.Address") ?? "ff15::1";
+                address = _communicator.GetProperty("IceDiscovery.Address") ?? "239.255.0.1";
```

Comment:
maybe we can do
```
string address = _communicator.GetProperty("IceDiscovery.Address") ??
    (_communicator.PreferIPv6 ? "ff15::1" : "239.255.0.1");
```

also remove `preferIPv6` local variable use `_communicator.PreferIPv6` directly